### ISSUE TITLE
Get-MSIntuneAuthToken fail saying it cannot find a dll

### ISSUE THREAD

**Issue by nodiaque** [MSEndpointMgr/ModernDriverManagement#268]
Hello,

The AzureAD powershell module removed the DLL Microsoft.IdentityModel.Clients.ActiveDirectory.dll in latest version. Because of that, when the psintune powershell module is install and the get-msintuneauthtoken is used, it failed saying it cannot find the required dll. I think this method is deprecated and a new way must be used to get the token.

**Comment by nodiaque** [comment]
Yeah, it seems modern driver management is dying

**Comment by MagnusE007** [comment]
Actually, there is nothing wrong with the MDM script itself, the problem lies in the module PSIntuneAuth (Nickolaj Andersen) which is used to authenticate with and which relies on the AzureAD module being on the device.

The PSIntuneAuth module downloads the latest AzureAD module which no longer contains the files (Microsoft.IdentityModel.Clients.ActiveDirectory.dll, Microsoft.IdentityModel.Clients.ActiveDirectory.Platform.dll, Microsoft.IdentityModel.Clients.ActiveDirectory.Platform.xml and Microsoft. IdentityModel.Clients.ActiveDirectory.xml) which are expected to be in that module. 

It would work if the PSIntuneAuth module was scripted to use a specific version of the AzureAD module that contained the files or updated to support Microsoft's latest AzureAD module.

I added a step in the task-sequence before the MDM script runs that checks if the computer is on the Internet and if so, downloads the AzureAD module and copies in the missing files that are expected to be there. Not the prettiest but it works!

**Comment by nodiaque** [comment]
Yeah I already opened a ticket on the PSIntuneAuth because it's using deprecated content, but I don't think it will be updated unfortunatly.
